File: LeetCode/LeetCode 75/1071. Greatest Common Divisor of Strings.java
class Solution {
    private int gcd(int a, int b) {
        if (b == 0){
            return a;
        }
        else{
            return gcd(b, a%b);
        }
    }

    public String gcdOfStrings(String str1, String str2) {
        int lenOne = str1.length();
        int lenTwo = str2.length();
        
        if(!(str1+str2).equals(str2+str1)) {
            return "";
        }

        int gcdLen = gcd(lenOne, lenTwo);
        return str1.substring(0, gcdLen);

    }
}

File: Leetcode/Leetcode 75/1071. Greatest Common Divisor of Strings.js
/**
 * @param {string} str1
 * @param {string} str2
 * @return {string}
 */
var gcdOfStrings = function(str1, str2) {
    if (str1 + str2 !== str2 + str1) return "";
    let shorterLen = Math.min(str1.length, str2.length);
    for (let i = shorterLen; i>0; i--) {
        if (str1.length % i === 0 && str2.length % i === 0) {
            return str1.substring(0, i);
        }
    }
    return "";

};
